packages/frontend/src/components/RichTextEditor/index.tsx

```diff
@@ -71,14 +99,44 @@ const Editor = ({
     // publish and unpublish of pipe
     editor?.setOptions({ editable })
   }, [editable])
+  const handleVariableClick = useCallback(
+    (variable: Variable) => {
+      editor?.commands.insertContent({
+        type: StepVariable.name,
+        attrs: {
+          id: variable.name,
+          label: variable.label,
+          value: variable.value,
+        },
+      })
+    },
+    [editor],
+  )
+
   return (
     <div className="editor">
       <MenuBar editor={editor} />
-      <EditorContent
-        className="editor__content"
-        editor={editor}
-        placeholder="Hello world"
-      />
+      <ClickAwayListener
+        mouseEvent="onMouseDown"
+        onClickAway={() => setShowVarSuggestions(false)}
+      >
+        <div ref={editorRef}>
+          <EditorContent
+            className="editor__content"
+            editor={editor}
+            placeholder="Hello world"
+            onFocus={() => setShowVarSuggestions(true)}
+          />
+          {variablesEnabled && (
+            <SuggestionsPopper
+              open={showVarSuggestions}
+              anchorEl={editorRef.current}
+              data={stepsWithVariables}
+              onSuggestionClick={handleVariableClick}
+            />
+          )}
```

Comment:
With reference to my [comment](https://github.com/opengovsg/plumber/pull/274/files#r1368524777) above

```suggestion
          {variablesEnabled && (
            <div
              onMouseEnter={() => {
                editor?.commands.blur()
              }}
            >
              <SuggestionsPopper
                open={showVarSuggestions}
                anchorEl={editorRef.current}
                data={stepsWithVariables}
                onSuggestionClick={handleVariableClick}
              />
            </div>
          )}
```

This hack will allow variable insertion without needing to click a 2nd time.  But it screws up because when you mouse over the popper, user cannot type inside the editor anymore.

Reply:
I think the "screws up" part would actually happen quite often, and is rather a downgrade in experience.